src/package.rs

```diff
@@ -262,88 +317,11 @@ impl PackageStore {
             .wrap_err(miette!("failed to finalize package"))?
             .into();
 
-        tracing::info!(
-            ":: packaged {}@{}",
-            manifest.package.name,
-            manifest.package.version
-        );
-
         Ok(Package::new(manifest, tgz))
     }
 
-    /// Directory for the vendored installation of a package
-    pub fn locate(package: &PackageName) -> PathBuf {
-        PathBuf::from(Self::PROTO_VENDOR_PATH).join(package.as_str())
-    }
-
-    /// Collect .proto files in a given path whilst excluding vendored ones
-    pub async fn collect(path: &Path) -> Vec<PathBuf> {
-        let vendor_path = fs::canonicalize(&Self::PROTO_VENDOR_PATH)
-            .await
-            .unwrap_or(Self::PROTO_VENDOR_PATH.into());
-
-        let mut paths: Vec<_> = WalkDir::new(path)
-            .into_iter()
-            .filter_map(Result::ok)
-            .map(|entry| entry.into_path())
-            .filter(|path| !path.starts_with(&vendor_path))
-            .filter(|path| {
-                let ext = path.extension().map(|s| s.to_str());
-
-                matches!(ext, Some(Some("proto")))
-            })
-            .collect();
-
-        paths.sort(); // to ensure determinism
-
-        paths
-    }
-}
-
-/// An in memory representation of a `buffrs` package
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Package {
-    /// Manifest of the package
-    pub manifest: Manifest,
-    /// The `tar.gz` archive containing the protocol buffers
-    pub tgz: Bytes,
-}
-
-impl Package {
-    /// Creates a new package
-    pub fn new(manifest: Manifest, tgz: Bytes) -> Self {
-        Self { manifest, tgz }
-    }
-
-    /// The name of this package
-    #[inline]
-    pub fn name(&self) -> &PackageName {
-        &self.manifest.package.name
-    }
-
-    /// The version of this package
-    #[inline]
-    pub fn version(&self) -> &Version {
-        &self.manifest.package.version
-    }
-
-    /// Lock this package
-    ///
-    /// Note that despite returning a Result this function never fails
-    pub fn lock(
-        &self,
-        registry: RegistryUri,
-        repository: String,
-        dependants: usize,
-    ) -> miette::Result<LockedPackage> {
-        LockedPackage::lock(self, registry, repository, dependants)
-    }
-}
-
-impl TryFrom<Bytes> for Package {
-    type Error = miette::Report;
-
-    fn try_from(tgz: Bytes) -> Result<Self, Self::Error> {
+    /// Load a package from a precompressed archive.
+    pub fn load(tgz: Bytes) -> miette::Result<Self> {
```

Comment:
Imo lets not do this, also we dont load, but rather parse

Reply:
I don't quite follow?

Reply:
So, you moved this from `TryFrom<Bytes>` into `::load` and i think there is no reason, as we dont reuse this, and the semantic meaning of load is weird – we dont do IO here

Reply:
Well, having any kind of complex logic inside of a trait implementation is a strong code smell — so I had to move this into a method. I could make the method non-public and rename it. But I would definitely not leave logic inside the `impl TryFrom<Bytes>`, that is not a good idea.

Reply:
I'll rename it to parse and make it non-pub perhaps.